pymongo/client_session.py

```diff
@@ -394,9 +394,10 @@ def commit_transaction(self):
             # We're rerunning the commit, move the state back to "in progress"
             # so that _in_transaction returns true.
             self._transaction.state = _TxnState.IN_PROGRESS
+            retry = True
```

Comment:
Let's also comment here that landing up here means the application has explicitly retried committing the transaction.